### PR TITLE
[KYUUBI-SHADED #60] Step 1/2: Copy TSaslServerTransport from Thrift 0.16.0

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

The first step of https://github.com/apache/kyuubi-shaded/issues/60, copy before making modifications to achieve a clear diff in the commit history.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-relocated-thrift/src/main/java/org/apache/thrift/transport/TSaslServerTransport.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+     * Get a new <code>TSaslServerTransport</code> instance, or reuse the existing one if a <code>
+     * TSaslServerTransport</code> has already been created before using the given <code>TTransport
+     * </code> as an underlying transport. This ensures that a given underlying transport instance
+     * receives the same <code>TSaslServerTransport</code>.
+     */
+    @Override
+    public TTransport getTransport(TTransport base) throws TTransportException {
+      WeakReference<TSaslServerTransport> ret = transportMap.get(base);
+      if (ret == null || ret.get() == null) {
+        LOGGER.debug("transport map does not contain key {}", base);
```
Comment (pan3793) [inline kyuubi-relocated-thrift/src/main/java/org/apache/thrift/transport/TSaslServerTransport.java:203]: this is a small fix by adding the missing placeholder `{}`

**Review by cxzl25 — APPROVED** [review]
